lld/MachO/DriverUtils.cpp

```diff
@@ -225,21 +225,42 @@ std::optional<StringRef> macho::resolveDylibPath(StringRef dylibPath) {
 // especially if it's a commonly re-exported core library.
 static DenseMap<CachedHashStringRef, DylibFile *> loadedDylibs;
 
+static StringRef realPathIfDifferent(StringRef path) {
+  SmallString<128> realPathBuf;
+  std::error_code err = fs::real_path(path, realPathBuf);
+  if (err)
+    return StringRef();
+
+  StringRef realPath(realPathBuf);
+  if (realPath.ends_with(path))
+    return StringRef();
```

Comment:
Is this suppose to detect symlinks? I don't think covers all cases. Imagine that in `/a/b/` you have a symlink `dir` to `/c/d/dir`. In the paths `dir` and `/c/d/dir`, the suffixes are the same, but they are different paths.

Reply:
The intent is to check for when the real_path output matches the input, yes. You are right that the check is insufficient, I suppose I will need to create an absolute path for the input too to compare.